extensions/teamup-rooms/src/index.tsx

```diff
@@ -0,0 +1,254 @@
+          },
+        };
+    });
+
+  return event;
+};
+
+const getSubcalendars = async (auth: GotConfig) => {
+  const { body } = await got(`${auth.apiRoot}/subcalendars`, {
+    headers: { ...auth.headers },
+  });
+
+  return body;
+};
+
+function TimeDropdown(props: { onTimeChange: (x: TimeObject) => void }) {
+  const { onTimeChange } = props;
+
+  return (
+    <List.Dropdown
+      tooltip="How long do you need a room?"
+      defaultValue={"30"}
+      onChange={(minutes) => {
+        onTimeChange(times.filter((time) => time.minutes === minutes)[0]);
+      }}
+    >
+      {times.map((time) => (
+        <List.Dropdown.Item key={time.minutes} title={time.text} value={time.minutes} />
+      ))}
+    </List.Dropdown>
+  );
+}
+
+export default function Command() {
+  const [openRooms, setOpenRooms] = useState<any>(undefined);
+  const [timeString, setTimeString] = useState<string>("30 minutes");
+  const [minutesString, setMinutesString] = useState<string>("30");
+  const [minutes, setMinutes] = useState<number>(30);
+
+  const startDt = new Date();
+  const endDt = new Date(startDt.getTime() + minutes * 60000);
+
+  const { calendar, token, default_title: defaultTitle } = getPreferenceValues<Preferences>();
+
+  const auth = createAuth({ calendar, token });
+
+  const quickAddEvent = async (roomId: Room["id"]) => {
+    const toast = await showToast({
+      style: Toast.Style.Animated,
+      title: "Booking room",
+    });
+
+    const event: any = await bookRoom(auth, roomId, minutes, defaultTitle);
+
+    if (event.error) {
+      toast.style = Toast.Style.Failure;
+      toast.title = event.error.title;
+      toast.message = event.error.message;
+    } else {
+      toast.hide();
+      setOpenRooms(openRooms.filter((room: Room) => room.id !== roomId));
+      await showHUD("🗓️ Room booked");
+    }
+  };
+
+  const loadRooms = async () => {
+    let events: any;
+    let rooms: any;
+    const occupiedRooms: any = [];
+
+    await getCalendar(auth).then((res) => (events = JSON.parse(res).events));
+    await getSubcalendars(auth).then((res) => (rooms = JSON.parse(res).subcalendars));
+
+    events = events
+      .map((event: any) => {
+        return {
+          ...event,
+          start_dt: new Date(event.start_dt),
+          end_dt: new Date(event.end_dt),
+        };
+      })
+      .filter((event: any) => event.start_dt < endDt)
+      .filter((event: any) => event.end_dt > startDt);
+
+    for (const ev in events) {
+      occupiedRooms.push(events[ev].subcalendar_id);
+    }
+
+    rooms = rooms.filter((room: any) => {
+      return occupiedRooms.indexOf(room.id) === -1;
+    });
+
+    setOpenRooms(rooms);
+  };
+
+  useEffect(() => {
+    setOpenRooms(undefined);
+    loadRooms();
+  }, [minutes]);
+
+  useEffect(() => {
+    if (minutesString) setMinutes(parseInt(minutesString));
+  }, [minutesString]);
+
+  return (
+    <List
+      navigationTitle={`Rooms open in the next ${timeString}`}
```

Comment:
```suggestion
```
Don't change the navigationTitle in the root command

Reply:
@pernielsentikaer No problem, I'll make this change. Has there been any discussion about whether a dynamic badge or feedback area could be added in the title area? It seems like a logical place for feedback since there isn't a design guideline for adding a heading or descriptive test in the list.

Reply:
@drewwatkins11 At the moment it's done in most of the cases with a <section title={}> but it's something that will be discuessed later on again since it's something that pops up from time to time. Forgot to mention where I [quoted from](https://raycastcommunity.slack.com/archives/C01AC2X0GMN/p1655440176400979?thread_ts=1655219827.794579&cid=C01AC2X0GMN) 😄